WordPress/Classes/ViewRelated/Post/Preview/PreviewWebKitViewController.swift

```diff
@@ -105,6 +164,69 @@ class PreviewWebKitViewController: WebKitViewController {
+            deviceLabel.rightAnchor.constraint(equalTo: view.safeRightAnchor, constant: 4),
+            deviceLabel.bottomAnchor.constraint(equalTo: view.safeBottomAnchor, constant: 4)
+        ])
+        showLabel(device: selectedDevice)
+    }
+
+    private func showLabel(device: PreviewDeviceSelectionViewController.PreviewDevice) {
+        deviceLabel.isHidden = device == .default
+        deviceLabel.text = device.title
+    }
+}
+
+// MARK: UIPopoverPresentationDelegate
+
+extension PreviewWebKitViewController {
+    override func prepareForPopoverPresentation(_ popoverPresentationController: UIPopoverPresentationController) {
+        guard let navigationController = navigationController, popoverPresentationController.presentedViewController is PreviewDeviceSelectionViewController else {
+            super.prepareForPopoverPresentation(popoverPresentationController)
+            return
+        }
+
+        popoverPresentationController.permittedArrowDirections = .down
+
+        popoverPresentationController.sourceRect = CGRect(x: navigationController.toolbar.frame.maxX - 36, y: navigationController.toolbar.frame.minY - 2, width: 0, height: 0)
+        popoverPresentationController.sourceView = navigationController.toolbar.superview
+    }
+
+    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
+        return .none
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        // Reset our source rect and view for a transition to a new size
+        guard let navigationController = navigationController,
+            let popoverPresentationController = presentedViewController?.presentationController as? UIPopoverPresentationController,
+            popoverPresentationController.presentedViewController is PreviewDeviceSelectionViewController else { return }
+
+        popoverPresentationController.sourceRect = CGRect(x: navigationController.toolbar.frame.maxX - 36, y: navigationController.toolbar.frame.minY - 2, width: 0, height: 0)
+        popoverPresentationController.sourceView = navigationController.toolbar.superview
+    }
+}
+
+// MARK: WKNavigationDelegate
+
+extension PreviewWebKitViewController {
+    func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
+        if selectedDevice == .desktop {
+            // Change the viewport scale to match a desktop environment
+            webView.evaluateJavaScript("let originalVp = document.querySelector('meta[name=viewport]').cloneNode(true); originalVp.setAttribute('name', 'original_viewport' ); document.querySelector('head').appendChild(originalVp); parent = document.querySelector('meta[name=viewport]'); parent.setAttribute('content','initial-scale=0');", completionHandler: nil)
+        }
+    }
 }
 
 // MARK: NoResultsViewController Delegate
```

Comment:
I know this wasn't added in this PR, but I just noticed - it doesn't look like this NRV has an action button, so this shouldn't be needed. Which means you can also remove `controller.delegate = self` when creating the `NoResultsViewController`.

Reply:
That's a good point. We're not using it currently for failed loads like we were in the old preview screen.